src/pybind/mgr/volumes/fs/operations/versions/subvolume_v1.py

```diff
@@ -768,7 +781,9 @@ def get_pending_clones(self, snapname):
 
         for track_id in pending_track_id_list:
             try:
-                link_path = self.fs.readlink(os.path.join(index_path, track_id), 4096)
+                path = os.path.join(index_path, track_id)
+                link_path = self.fs.readlink(path, 4096)
+                self.clean_dangling_symlink(path, link_path, track_id)
             except cephfs.Error as e:
                 if e.errno != errno.ENOENT:
                     raise VolumeException(-e.args[0], e.args[1])
```

Comment:
@rishabh-d-dave the idea is to have a dedicated function for the removing non-existing pending clones. That organises the code cleanly and improve code readability. This would defeat that purpose. This would split the responsibility to caller and the callee function (clean_dangling_symlink_for_clone)